### PR TITLE
Flow Type Fixes Regarding Categories

### PR DESCRIPTION
**Pull request by dakeyras7**
Some a bit more complex fixes through 'Type Refinement'. Some others were just missing types or wrong types. Also I cleaned up any linting errors that I encountered.

### CODE REVIEW

**Comment by opyh** [comment]
Cool stuff. Will merge this soon.

File: src/components/NodeToolbar/NodeHeader.js [hunk 1]
```diff
@@ -70,7 +74,7 @@ export default class NodeHeader extends React.Component<Props> {
     const { category, parentCategory } = this.props;
     const shownCategory = category || parentCategory;
     let categoryName = shownCategory && categoryNameFor(shownCategory);
-    const shownCategoryId = shownCategory && shownCategory._id;
+    const shownCategoryId = shownCategory && getCategoryId(shownCategory);
```
Comment (opyh) [inline src/components/NodeToolbar/NodeHeader.js]: 👍

File: src/lib/Categories.js [hunk 2]
```diff
@@ -12,22 +12,37 @@ import type {
...
+  exact object types, and it's used here on purpose. Without it it's tough for
+  Flow to determine the actual specific type when it sees the `Category` union
+  type in parts of our codebase.
+
+  This is a pretty "debatable" feature of Flow and we should be aware of this
+  when we migrate our code to TypeScript in the future. Flow's own explanation
+  of why we need this can be found here:
+  https://flow.org/en/docs/types/unions/#toc-disjoint-unions-with-exact-types
+  https://flow.org/en/docs/lang/width-subtyping/
+*/
```
Comment (opyh) [inline src/lib/Categories.js]: 👍

**Comment by dakeyras7** [comment]
Will do

**Pull request merged**